impeller/display_list/aiks_dl_basic_unittests.cc

```diff
@@ -0,0 +1,768 @@
+  };
+
+  paint.setColor(DlColor::kBlue());
+  draw(paint, 0, 0);
+
+  DlColor gradient_colors[7] = {
+      DlColor::RGBA(0x1f / 255.0, 0.0, 0x5c / 255.0, 1.0),
+      DlColor::RGBA(0x5b / 255.0, 0.0, 0x60 / 255.0, 1.0),
+      DlColor::RGBA(0x87 / 255.0, 0x01 / 255.0, 0x60 / 255.0, 1.0),
+      DlColor::RGBA(0xac / 255.0, 0x25 / 255.0, 0x53 / 255.0, 1.0),
+      DlColor::RGBA(0xe1 / 255.0, 0x6b / 255.0, 0x5c / 255.0, 1.0),
+      DlColor::RGBA(0xf3 / 255.0, 0x90 / 255.0, 0x60 / 255.0, 1.0),
+      DlColor::RGBA(0xff / 255.0, 0xb5 / 255.0, 0x6b / 250.0, 1.0),
+  };
+  Scalar stops[7] = {
+      0.0,
+      (1.0 / 6.0) * 1,
+      (1.0 / 6.0) * 2,
+      (1.0 / 6.0) * 3,
+      (1.0 / 6.0) * 4,
+      (1.0 / 6.0) * 5,
+      1.0,
+  };
+  auto texture = CreateTextureForFixture("airplane.jpg",
+                                         /*enable_mipmapping=*/true);
+  auto image = DlImageImpeller::Make(texture);
+
+  paint.setColorSource(DlColorSource::MakeRadial(
+      {500, 600}, 75, 7, gradient_colors, stops, DlTileMode::kMirror));
+  draw(paint, 0, 300);
+
+  DlImageColorSource image_source(image, DlTileMode::kRepeat,
+                                  DlTileMode::kRepeat,
+                                  DlImageSampling::kNearestNeighbor);
+  paint.setColorSource(&image_source);
+  draw(paint, 300, 0);
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, CanSaveLayerStandalone) {
+  DisplayListBuilder builder;
+
+  DlPaint red;
+  red.setColor(DlColor::kRed());
+
+  DlPaint alpha;
+  alpha.setColor(DlColor::kRed().modulateOpacity(0.5));
+
+  builder.SaveLayer(nullptr, &alpha);
+
+  builder.DrawCircle({125, 125}, 125, red);
+
+  builder.Restore();
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, CanRenderDifferentShapesWithSameColorSource) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+
+  DlColor colors[2] = {
+      DlColor::RGBA(0.9568, 0.2627, 0.2118, 1.0),
+      DlColor::RGBA(0.1294, 0.5882, 0.9529, 1.0),
+  };
+  DlScalar stops[2] = {
+      0.0,
+      1.0,
+  };
+
+  paint.setColorSource(DlColorSource::MakeLinear({0, 0}, {100, 100}, 2, colors,
+                                                 stops, DlTileMode::kRepeat));
+
+  builder.Save();
+  builder.Translate(100, 100);
+  builder.DrawRect(SkRect::MakeXYWH(0, 0, 200, 200), paint);
+  builder.Restore();
+
+  builder.Save();
+  builder.Translate(100, 400);
+  builder.DrawCircle({100, 100}, 100, paint);
+  builder.Restore();
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, CanRenderRoundedRectWithNonUniformRadii) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+  paint.setColor(DlColor::kRed());
+
+  SkRRect rrect;
+  SkVector radii[4] = {
+      SkVector{50, 25},
+      SkVector{25, 50},
+      SkVector{50, 25},
+      SkVector{25, 50},
+  };
+  rrect.setRectRadii(SkRect::MakeXYWH(100, 100, 500, 500), radii);
+
+  auto path = SkPath::RRect(rrect);
+
+  builder.DrawPath(path, paint);
```

Comment:
Should this call builder.DrawRRect() here instead of DrawPath()? The AIKS test didn't have an AiksCanvas::DrawRRect() to call so it was written using PathBuilder.AddRoundedRect(), but builder does have a DrawRRect (which takes RRects with non-uniform radii)...

Reply:
I went with this option since it actually eliminates a golden file mis-match here. (One less impact on the goldens.)